src/qcodes/instrument_drivers/stahl/stahl.py

```diff
@@ -144,14 +147,32 @@ class Stahl(VisaInstrument):
 
     Args:
         name
-        address: A serial port address
+        address: A serial port address or TCP-IP visa address
+
+    The TCP/IP scenario can be used when the Stahl is connected to
+    a different computer for example a Raspberry Pi running Linux and exposed
+    using something like the following script:
+
+    ```
+    #!/bin/sh
+    DEVICE=/dev/ttyUSB0
+    PORT=8088
+    echo Listening...
+    while socat $DEVICE,echo=0,b115200,raw tcp-listen:$PORT,reuseaddr,nodelay; do
+        echo Restarting...
+    done
+    ```
+
+    In this case the address would be: `TCPIP0::hostname::8088::SOCKET`
     """
 
     def __init__(self, name: str, address: str, **kwargs: Any):
         super().__init__(name, address, terminator="\r", **kwargs)
-        assert isinstance(self.visa_handle, SerialInstrument)
-
-        self.visa_handle.baud_rate = 115200
+        if isinstance(self.visa_handle, TCPIPSocket):
+            pass # allow connection to remote serial device
+        else:
```

Comment:
```suggestion
        elif isinstance(self.visa_handle, SerialInstrument):
            self.visa_handle.baud_rate = 115200
        else:
            raise TypeError("VisaHandle must be either a 'SerialInstrument' or a 'TCPIPSocket')
```
I think its cleaner to make the error explicit now

Reply:
on it...